fcc verification progress on server

no-stack-dub-sack opened the pull request:
@stephenjfox @Bigghead @bonham000 Ok, Shav and I made some progress this evening on verifying users eligibility to join based on certs earned, please see comment on line 58 or `userRoute.js` though. Still having a little bit of trouble getting the right data to the client. 

I'm positive the update in the DB is happening as it should because I'm also logging the user on sign in, and when you sign in next, it shows the correctly updated data, it's just not being returned right after it updates correctly like we need it to. Shav's idea from the Slack chat could work well if we can get it working, but I would prefer the data coming directly from the DB.

Also, once we figure this out, we still need to:
 - redirect on failed verification to message that says why they can't join
 - set some kind of token that prevents that route from running again once they've been verified
 - redirect to a true welcome / profile setup page on successful verification

no-stack-dub-sack commented on backend/routes/userRoute.js:
```diff
@@ -17,11 +24,46 @@ router.get('/api/user', (req, res) => {
+    // even though user is already updated by the time of this
+    // .then(), what's returned is the document before the update
+    // so info is not current, we need this data to be current
+    // seems like it should work, not sure why it doesn't 
```
☝️ here's the issue we're having

bonham000 replied:
I'm not sure exactly why you wouldn't be getting the updated user at the end of this... but why not just create one array/object based on the resolution of these three calls to FCC and then find & save the user once? As in update something based on the redirect counts to produce something looking like this:

```
var certs = {
   front: true,
   data: false,
   back: true
}
```

and then save the user and return, e.g.

```
user.fccCerts = certs;
user.save();
res.json({ user });
```

no-stack-dub-sack replied:
@bonham000 this is similar to what Shave suggested yesterday on the slack chat, except we were trying to create the object outside of the axios call and get it to resolve the that object to we could then return it to the client. This does seem like a better way to save the data to the database though, and if the object is created inside the axios.all call, it should't be a problem. I / we (whoever is up to pair tonight) can update this PR instead or merging it, then fixing

bonham000 commented:
The user is not updated because of the order in which the promises execute. The `.then` chained onto the `axios.spread` is called before the mongoose calls within it update in the database.

This logs each `if` block in the `axios.spread` when we find a `cert` and when the first one is saved and also when the last `.then` is called. The last `.then` comes before the first is saved.

![screen shot 2017-03-06 at 10 36 01 pm](https://cloud.githubusercontent.com/assets/18126719/23644719/7e94945a-02bd-11e7-9e44-4e9a9f87069d.png)

no-stack-dub-sack commented:
@bonham000 Ahhh, the power of `console.log()`, I should have thought of this. It's weird though, I thought that chained then shouldn't be called until what's inside the previous promise resolves, but I'm still learning about all this promise stuff

no-stack-dub-sack commented:
Blocked for merge - will work with whoever wants to pair tonight to update PR with @bonham000's sensible suggestions. Similar to Shav's idea from yesterday.

no-stack-dub-sack commented:
Ok, I updated this PR and squashed my commits to keep our commit history nice and neat. Implemented something similar to sean's solution. Also, updated user schema to include GH data, and update the DB either way in our verification route. Since by this time the user already exists in our DB, we need as easy way to determine if they're verified, so added a `verifiedUser` key and update that if no certs and if certs (boolean)
